<?xml version="1.0"?>
<launch>
	<node pkg="tf"  type="static_transform_publisher" name="camera_to_origin" args="0.2 0 0.43 0.6877 -0.6877 -0.1644 0.1644 origin camera 20" />
	<node pkg="usb_cam" type="usb_cam_node" name="camera" output="screen">
		<param name="camera_info_url" type="string" value="file://$(find masterslave)/camera_info/head_camera_hand_720p_new_checkerboard2.yaml" />
		<param name="video_device" type="string" value="/dev/v4l/by-id/usb-Microsoft_Microsoft®_LifeCam_Cinema_TM_-video-index0"/>
		<param name="image_height" value="720" />
		<param name="image_width" value="1280" />
		<param name="camera_frame_id" type="string" value="/camera" />
		<param name="framerate" value="60" />
		<param name="brightness" value="20" />
		<param name="focus" value="9" />
		<param name="sharpness" value="180"/>
	</node>
	<node ns="camera" pkg="image_proc" type="image_proc" name="image_proc_node" output="screen"/>
		
	<arg name="marker_size_hand" default="1.4" />
      	<arg name="max_new_marker_error_hand" default="0.1" /> 
        <arg name="max_track_error_hand" default="0.05" />

      
        <arg name="cam_image_topic" default="/camera/image_mono" />
        <arg name="cam_info_topic" default="/camera/camera_info" />   
        <arg name="output_frame" default="/origin" />
	<arg name="med_filt_size" default="10" />
	<arg name="bundle_files_hand" default="$(find masterslave)/bundles/Thumb16.xml $(find masterslave)/bundles/IndexFinger16.xml"/>
	


        <node ns="hand" name="ar_track_alvar" pkg="ar_track_alvar" type="findMarkerBundlesNoKinect" respawn="false" output="screen" args="$(arg marker_size_hand) $(arg max_new_marker_error_hand) $(arg max_track_error_hand) $(arg cam_image_topic) $(arg cam_info_topic) $(arg output_frame) $(arg bundle_files_hand)" />
	
	<node pkg="masterslave" name="MasterSlaveManipulationAbsolute" type="MasterSlaveManipulationAbsolute" respawn="false" output="screen"/>	
	
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find masterslave)/rviz/UrsulaRviz.rviz" />
</launch>
